custom_components/iec/coordinator.py

```diff
@@ -568,17 +569,45 @@ async def _insert_statistics(self, contract_id: int, is_smart_meter: bool) -> No
 
     @staticmethod
     def _calculate_estimated_bill(meter_id, future_consumptions: dict[str, FutureConsumptionInfo | None], kwh_tariff,
-                                  kva_tariff, distribution_tariff, delivery_tariff, power_size):
+                                  kva_tariff, distribution_tariff, delivery_tariff, power_size, last_invoice):
         future_consumption_info: FutureConsumptionInfo = future_consumptions[meter_id]
         future_consumption = future_consumption_info.future_consumption
 
-        now = datetime.now()
-        days_of_invoice = now.day
-        days_in_current_month = calendar.monthrange(now.year, now.month)[1]
+        kva_price = power_size * kva_tariff / 365
+
+        total_kva_price = 0
+        distribution_price = 0
+        delivery_price = 0
 
         consumption_price = future_consumption * kwh_tariff
-        kva_price = power_size * kva_tariff / 365
-        distribution_price = (distribution_tariff / days_in_current_month) * days_of_invoice
-        delivery_price = (delivery_tariff / days_in_current_month) * days_of_invoice
 
-        return consumption_price + kva_price + distribution_price + delivery_price
+        if last_invoice != EMPTY_INVOICE:
```

Comment:
@tzagim - WDYT?

Reply:
After yesterday's debug, it seems that the date of the invoice has no meaning or at least doesn't have the same effect as from_date.

But I still think this change is good.